lib/models.js

```diff
@@ -0,0 +1,223 @@
+/** @babel */
+
+import fs from 'fs';
+import stdPath from 'path';
+
+import mkdirp from 'mkdirp';
+import touch from 'touch';
+
+import {DEFAULT_ACTIVE_FILE_DIR, DEFAULT_PROJECT_ROOT} from './config';
+import {
+    absolutify,
+    cachedProperty,
+    defineImmutable,
+    getProjectPath,
+    preferredSeparatorFor
+} from './utils';
+
+
+/**
+ * Wrapper for dealing with filesystem paths.
+ */
+export class Path {
+    constructor(path='') {
+        // The last path segment is the "fragment". Paths that end in a
+        // separator have a blank fragment.
+        let sep = preferredSeparatorFor(path);
+        let parts = path.split(sep);
+        let fragment = parts[parts.length - 1];
+        let directory = path.substring(0, path.length - fragment.length);
+
+        // Set non-writable properties.
+        defineImmutable(this, 'directory', directory);
+        defineImmutable(this, 'fragment', fragment);
+        defineImmutable(this, 'full', directory + fragment);
```

Comment:
Is this equivalent to `defineImmutable(this, 'full', path)`? If so, I think that would be a better way to word it.